src/dos/drive_fat.cpp

```diff
@@ -256,25 +256,30 @@ bool fatFile::Write(uint8_t * data, uint16_t *size) {
 }
 
 bool fatFile::Seek(uint32_t *pos, uint32_t type) {
-	int32_t seekto=0;
-	
+	int32_t requested_pos = 0;
+
 	switch(type) {
-		case DOS_SEEK_SET:
-			seekto = (int32_t)*pos;
-			break;
-		case DOS_SEEK_CUR:
-			/* Is this relative seek signed? */
-			seekto = (int32_t)*pos + (int32_t)seekpos;
-			break;
-		case DOS_SEEK_END:
-			seekto = (int32_t)filelength + (int32_t)*pos;
-			break;
+	case DOS_SEEK_SET: requested_pos = (int32_t)*pos; break;
+	case DOS_SEEK_CUR:
+		// Is this relative seek signed?
+		requested_pos = (int32_t)*pos + (int32_t)seekpos;
+		break;
+	case DOS_SEEK_END:
+		requested_pos = (int32_t)filelength + (int32_t)*pos;
+		break;
 	}
 //	LOG_MSG("Seek to %d with type %d (absolute value %d)", *pos, type, seekto);
 
-	if(seekto<0) seekto = 0;
-	seekpos = (uint32_t)seekto;
+	if (requested_pos < 0) {
+		requested_pos = 0;
+	}
+
+	if (seekpos != static_cast<uint32_t>(requested_pos)) {
+		seekpos = static_cast<uint32_t>(requested_pos);
+	} 
```

Comment:
Same comment about the if statement.  This code does use `requested_pos` to check for wrapping so makes sense to have the variable.